File: GCJ/2016/QualificationRound/QualB.cpp
//https://code.google.com/codejam/contest/6254486/dashboard#s=p1
#include <iostream>
#include <string>
using namespace std;

int N;
string S;
bool debug = false;

void input() {
	cin >> S;
}

bool finish() {
	for (int i = 0; i < S.size(); ++i) {
		if (S[i] == '-') return false;
	}
	return true;
}

void solve() {
	int res = 0;
	while(!finish()) {
		int nFlip = 0;
		while (S[nFlip] == S[0]) {
			nFlip++;
		}
		string STemp;
		for (int i = 0; i < nFlip; ++i) {
			STemp.push_back('+' + '-' - S[nFlip - i - 1]);
		}
		for (int i = nFlip; i < S.size(); ++i) {
			STemp.push_back(S[i]);
		}
		S = STemp;
		res++;
	}
	cout << res << endl;
}

int main() {
	int T;
	cin >> T;
	for (int i = 1; i <= T; ++i) {
		input();
		cout << "Case #" << i << ": ";
		solve();
	}
}

File: GCJ/2016/QualificationRound/QualB.py
#https://code.google.com/codejam/contest/6254486/dashboard#s=p1

def solve():
	S = str(raw_input())
	res = 0
	for i in xrange(1,len(S)):
		if S[i] != S[i-1]:
			res += 1
	if S[len(S)-1] == '-':
		res += 1
	return res

def main():
	T = int(raw_input())
	for i in xrange(1,T+1):
		print "Case #{}: {}".format(i, solve())

if __name__=="__main__":
	main()
